### DISCUSSION TITLE
Visual Studio Debug IBM i

### DISCUSSION THREAD

**Discussion by rdhill10** [codefori/vscode-ibmi#2170] [IBM i] [answered]
Trying to setup the IBM i debug functionality in Visual Studio Code

The server job portion is running (under Visual Studio Code)  but having issues starting the service

The first error I was receiving as a spool file on AS400 side was Java version 11 is required.

So I have JDK11 option 19 installed now and I have it global now so JDK11 is the default.

Even with this install I'm still receiving a spool file on AS400 saying Java Version 11 is required.

Can someone point me to my issue?      I have verified I have the required PTFS SI86178 and SI80364/SI81031.

We are at V7R4 CUMULATIVE PTF PACKAGE C1091740

Thanks for any help

PS.....   Here is the error straight from the spool file

openjdk version "11.0.8" 2020-07-14                                                                                               
OpenJDK Runtime Environment 11.0.8.0-IBM (build 11.0.8+10)                                                                        
Eclipse OpenJ9 VM 11.0.8.0-IBM (build openj9-0.21.0, JRE 11 OS/400 ppc64-64-Bit Compressed References 20210119_000000 (JIT enabled
OpenJ9   - 34cf4c0                                                                                                                
OMR      - 113e542                                                                                                                
JCL      - 95bb504 based on jdk-11.0.8+10)                                                                                        
Incorrect Java version. Java 11 is required.

**Comment by chrjorgensen** [comment, marked as answer]
@rdhill10 Your Java11 version is quite old - please install the Java PTF group SF99665, the current level is 23.
After PTF install, check the Java level with command `runjva *version` - our system reports
```
java version "11.0.23" 2024-04-16                                                                                                
IBM Semeru Runtime Certified Edition (build 11.0.23+9)                                                                           
Eclipse OpenJ9 VM (build openj9-0.44.0, JRE 11 OS/400 ppc64-64-Bit Compressed References 20240609_000000 (JIT enabled, AOT enable
d)                                                                                                                               
OpenJ9   - b0699311c7                                                                                                            
OMR      - 254af5a04                                                                                                             
JCL      - 656fa737c3 based on jdk-11.0.23+9)                                                                                    
Java program completed                                                                                                           
```

**Reply by rdhill10** [reply]
@chrjorgensen  Thank You So Much For The Reply - I will get this scheduled .     Have a good day.

**Reply by chrjorgensen** [reply]
@rdhill10 You're welcome - and while you're at it, please update all(!) PTF groups! Your cum package is three years old...
Using the SQL view [SYSTOOLS.GROUP_PTF_CURRENCY](https://www.ibm.com/support/pages/node/1126971) would be a good start. 😃
